linalg/sundials.cpp

```diff
@@ -187,11 +187,21 @@ SundialsMemHelper &Sundials::GetMemHelper()
 Sundials::Sundials()
 {
 #ifdef MFEM_USE_MPI
-   MPI_Comm communicator = MPI_COMM_WORLD;
+   int mpi_initialized = 0;
+   MPI_Initialized(&mpi_initialized);
+   MPI_Comm communicator = mpi_initialized ? MPI_COMM_WORLD : MPI_COMM_NULL;
```

Comment:
As I understand it (which may not be correct), `communicator` will be `MPI_COMM_NULL` if the user calls `Sundials::Init()` before `Mpi::Init()`.  What is the behavior then within SUNDIALS with `context` created with `communicator = MPI_COMM_NULL`?

Reply:
In that case, any logging messages will be written from all ranks rather than just the root rank. @balos1 is there anything else I'm forgetting?

Reply:
> is there anything else I'm forgetting?

The sundials profiler will not be MPI-aware if the communicator is `MPI_COMM_NULL` (so the profiler output will happen on all ranks).

Reply:
Ok, sounds like all the behavior differences are limited to output being on all ranks instead of the root rank (i.e., no behavior change with solvers within SUNDIALS)?  @v-dobrev is this the intended use of `MPI_COMM_NULL`?

Reply:
The reason for this check is that the serial examples were failing in the SUNDIALS context creation -- that call was trying to use `MPI_COMM_WORLD` and duplicate it which was failing when MPI was not initialized. So the idea is:
* If running a serial example (with MPI enabled SUNDIALS build), MPI is not initialized, so we'll use `MPI_COMM_NULL` which seems to be fine based on the comments above,
* If running in parallel, MPI must be initialized before the `mfem::Sundials` singleton is created, i.e. before calling `Sundials::Init` or invoking and SUNDIALS classes in MFEM that construct the `mfem::Sundials` singleton.

So overall, I think we're fine with this logic.